src/contracts/pods/EigenPod.sol

```diff
@@ -443,30 +542,75 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
         external
         onlyEigenPodManager
     {
-        // reduce the restakedExecutionLayerGwei
-        restakedExecutionLayerGwei -= uint64(amountWei / GWEI_TO_WEI);
-
         emit RestakedBeaconChainETHWithdrawn(recipient, amountWei);
-
         // transfer ETH from pod to `recipient`
         _sendETH(recipient, amountWei);
     }
 
+
+    /**
+     * @notice Called by the pod owner to activate restaking by withdrawing 
+     * all existing ETH from the pod and preventing further withdrawals via 
+     * "withdrawBeforeRestaking()"
+    */ 
+    function activateRestaking() external onlyWhenNotPaused(PAUSED_EIGENPODS_VERIFY_CREDENTIALS) onlyEigenPodOwner hasNeverRestaked {
+        hasRestaked = true;
+        _processWithdrawalBeforeRestaking(podOwner);
+    }
```

Comment:
do we want this function to emit a `RestakingActivated` or similar event?
@gpsanant any thoughts on this?

Reply:
Agreed, added